src/Riok.Mapperly/Descriptors/Mappings/UserMappings/UserDefinedNewInstanceGenericTypeMapping.cs

```diff
@@ -35,7 +37,9 @@ ITypeSymbol objectType
     public override MethodDeclarationSyntax BuildMethod(SourceEmitterContext ctx)
     {
         var methodSyntax = (MethodDeclarationSyntax)Method.DeclaringSyntaxReferences.First().GetSyntax();
-        return base.BuildMethod(ctx).WithTypeParameterList(methodSyntax.TypeParameterList);
+        return base.BuildMethod(ctx)
+            .WithTypeParameterList(methodSyntax.TypeParameterList)
+            .WithConstraintClauses(List(GetTypeParameterConstraintClauses()));
```

Comment:
Is there a reason for not simply using `methodSyntax.ConstraintClauses` here?

Reply:
Yes, it modifies types to by `FullyQualifiedIdentifier`:
![image](https://github.com/riok/mapperly/assets/3594540/ccd23624-aa5b-4b43-9fd9-568ec95c2e37)